### ISSUE TITLE
Menus with options Observable keep old values

### ISSUE THREAD

**Issue by tecosaur** [JuliaPlots/Makie.jl#1956]
If I construct a Menu with an options Observable and update it to *less* values, some of the old values are still shown.

#### MWE

```julia
# Makie 0.6.2, GLMakie 0.17.2
fig = Figure()
opt = Observable(["A", "B", "C"])
Menu(fig[1,1], options=opt)
opt[] = ["D"]
```

![image](https://user-images.githubusercontent.com/20903656/169185460-f068b15a-e095-4785-aea8-70eed70ccc3c.png)

**Comment by tecosaur** [comment]
Extra info: This seems to only occur when `fig` is shown *prior* to setting `opt[] = ["D"]`. If `fig` is (re)displayed after, it works as expected.

**Comment by jkrumbiegel** [comment]
This is probably due to the additional scene nesting in Blocks since 0.17, which messed with the deletion behavior, because the menu logic hasn't changed.

**Comment by tecosaur** [comment]
Could that change also be related to the fact I'm now seeing truncate UInt8 errors with scene IDs now? (an issue I have yet to make an Issue about).

**Comment by jkrumbiegel** [comment]
Yes that is also related. If you have a lot of Blocks there are too many scenes created for GLMakie to handle. It wasn't caught at the time because our tests don't seem to use 255+ Scenes.

**Comment by tecosaur** [comment]
Ah, that's interesting. Could the fact so many scenes are generated also be related to it being rather slow to interact with? It's a scatter + contour + 2x density plot graph on ~2000 2d data points with several Menus and an IntervalSlider controlling the data.

**Comment by jkrumbiegel** [comment]
`Menu` could probably use a redesign that doesn't use `Label` and `Box` internally for the entries as those are probably too heavy. I'm not sure, @SimonDanisch said that Scenes should be very lightweight so that it shouldn't matter much that more are added

**Comment by ffreyer** [comment]
My impression is that interactivity is usually slow when it ends up triggering lots of things. For example in #1835 I made some improvements to Axis3 interactions. When rotating the axis the rough chain of observables for labels was:

1. mouse events trigger azimuth and elevation (probably mostly both)
2. azimuth (always) triggers mi1 and mi2, elevation (always) triggers mi3 (these hold information of the viewing angle, i.e. above or below, left or right, front or back) and both trigger an update to projectionview (amongst other matrices)
3. each of those trigger text position, alignment and rotation updates for aixs labels
4. And each of those triggers a full recalculation of text layouting

So starting from one mouse event you ended up triggering 1 -> 2 -> 5 -> 15 recalculation of the character layout for labels. Times 3 for x, y, z label and I also skipped ticks, grid lines and frame lines. 

After the pr mi1, mi2, mi3 only trigger when they actually change (which is rare), text positions no longer trigger a recalculation of the character layout and text alignment and rotation have been merged into one trigger. So the common chain has been reduced to 1 -> 2 -> 2 (up to 5) -> 2 (5) recalculations of the character layout.

**Comment by jkrumbiegel** [comment]
Yes you're completely right. I find it's not so easy to predict the best structure for an observable chain before I've written it once in its full form. For example, with Axis3, it was natural for me to think of azimuth and elevation separately, but of course in terms of observables it makes more sense to have one orientation value for the Axis and not split it..

**Comment by ffreyer** [comment]
I fell like a lot of times readability and usability are at odds with performance when using Observables. 

For example, changing `output = map(f, input)` to 
```julia
output = Observable(initital_value)
on(x -> output.val = x, input)
```
is often useful to avoid triggering downstream updates multiple times, but it takes more space and gives you no chance to follow the observable chain upwards.

Or patterns like
```julia
o1 = map(prep1, input)
o2 = map(prep2, input)
o3 = map(prep3, input)
output1 = map(f, o1, o2)
output2 = map(g, o1, o3)
```
which come up often and feel natural, but result in multiple triggers. Doing
```julia
output1 = map(x -> f(prep1(x), prep2(x)), input)
output2 = map(x -> g(prep1(x), prep3(x)), input)
```
would call `prep1` multiple times, which might be more expensive. So you'd have to do
```julia
o1 = map(prep1, input)
o2 = map(prep2, input)
o3 = map(prep3, input)
output1 = map(_ -> f(o1[], o2[]), input)
output2 = map(_ -> g(o1[], o3[]), input)
```
which makes it difficult to understand how updates flow.

**Comment by jkrumbiegel** [comment]
I still hope we can come up with a bundle-of-observables object that can help with this mess by grouping updates together and forwarding only to those places that need it.

**Comment by tecosaur** [comment]
The original issue raised here has now been solved in #2074.
